launcher/src/main/java/org/wso2/carbon/launcher/CarbonServerEvent.java

```diff
@@ -76,4 +96,17 @@ public int getType() {
     public CarbonLaunchConfig getConfig() {
         return config;
     }
+
+    /**
+     * returns the OSGi system bundle context.
+     * @return system bundle context
+     */
+    public BundleContext getSystemBundleContext() {
+        return systemBundleContext;
+    }
+
+    // Sets the system bundle context for the CarbonServerEvent
+    void setSystemBundleContext(BundleContext systemBundleContext) {
+        this.systemBundleContext = systemBundleContext;
+    }
```

Comment:
The getSystemBundleContext() is can used by the external listener implementation to access the system bundle. For example in [here ](https://github.com/Mirage20/Carbon-SPI/blob/master/carbon-spi-hook/src/main/java/org/wso2/osgi/spi/WeavingServiceHook.java#L22) the carbon-spi module uses system bundle for register an osgi weaving hook.